File: benchmark/ocaml/task5_class.ml
type point = { x : float; y : float }

let make_point x y = { x; y }

let distance_to p1 p2 =
  let dx = p2.x -. p1.x in
  let dy = p2.y -. p1.y in
  sqrt (dx *. dx +. dy *. dy)

File: benchmark/lua/task5_class.lua
Point = {}
Point.__index = Point

function Point.new(x, y)
    return setmetatable({x = x, y = y}, Point)
end

function Point:distance_to(other)
    local dx = self.x - other.x
    local dy = self.y - other.y
    return math.sqrt(dx * dx + dy * dy)
end
